mirrord-layer/src/tcp_mirror.rs

```diff
@@ -177,4 +178,41 @@ impl TcpHandler for TcpMirrorHandler {
     fn ports_mut(&mut self) -> &mut HashSet<Listen> {
         &mut self.ports
     }
+
+    async fn handle_listen_close(
+        &mut self,
+        close: ListenClose,
+        codec: &mut actix_codec::Framed<
+            impl tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send,
+            ClientCodec,
+        >,
+    ) -> Result<(), LayerError> {
+        codec
+            .send(ClientMessage::Tcp(LayerTcp::PortUnsubscribe(close.port)))
+            .await
+            .map_err(From::from)
+    }
+
+    async fn handle_listen(
```

Comment:
handle_listen and handle_listen_close are pretty much the same in tcp_steal and tcp_mirror, I'd try to deduplicate with generics